govcd/vdc.go

```diff
@@ -837,3 +836,41 @@ func (vdc *Vdc) buildNsxvNetworkServiceEndpointURL(optionalSuffix string) (strin
 
 	return hostname, nil
 }
+
+// GetEdgeGatewayReferenceList retrieves a list of edge gateway references.
+// Unlike other entities under Vdc, edge gateway are not listed in the parent entities
+// and therefore we need to get the list using a query.
+func (vdc *Vdc) GetEdgeGatewayReferenceList(refresh bool) ([]*types.Reference, error) {
```

Comment:
what is difference between `getEdgeGatewayRecordsType` and this function?

Reply:
The difference is only the return type.
I think I can achieve the same result by making `getEdgeGatewayRecordsType` public

Reply:
Or making more like wrapper function which converts to your needed type, but it's up to you.